fuzz/fuzz.go

```diff
@@ -0,0 +1,410 @@
+package fuzz
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/golang/mock/gomock"
+	"io"
+	"math"
+	"net/netip"
+	"testing"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/compiler"
+	"github.com/brimdata/zed/compiler/data"
+	"github.com/brimdata/zed/compiler/optimizer"
+	"github.com/brimdata/zed/compiler/optimizer/demand"
+	"github.com/brimdata/zed/compiler/semantic"
+	"github.com/brimdata/zed/pkg/nano"
+	"github.com/brimdata/zed/pkg/storage/mock"
+	"github.com/brimdata/zed/runtime"
+	"github.com/brimdata/zed/zbuf"
+	"github.com/brimdata/zed/zcode"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zio/vngio"
+	"github.com/brimdata/zed/zio/zngio"
+	"github.com/brimdata/zed/zson"
+	"github.com/stretchr/testify/require"
+	"github.com/x448/float16"
+)
+
+func ReadZNG(bs []byte) ([]zed.Value, error) {
+	bytesReader := bytes.NewReader(bs)
+	context := zed.NewContext()
+	reader := zngio.NewReader(context, bytesReader)
+	defer reader.Close()
+	var a zbuf.Array
+	err := zio.Copy(&a, reader)
+	if err != nil {
+		return nil, err
+	}
+	return a.Values(), nil
+}
+
+func ReadVNG(bs []byte) ([]zed.Value, error) {
+	bytesReader := bytes.NewReader(bs)
+	context := zed.NewContext()
+	reader, err := vngio.NewReader(context, bytesReader)
+	if err != nil {
+		return nil, err
+	}
+	var a zbuf.Array
+	err = zio.Copy(&a, reader)
+	if err != nil {
+		return nil, err
+	}
+	return a.Values(), nil
+}
+
+func WriteZNG(t *testing.T, valuesIn []zed.Value, buf *bytes.Buffer) {
```

Comment:
Nits:
1. Could use io.Writer here to make this a little more flexible.
2. We usually put the destination parameter first (as with zio.Copy).
3. We usually use val as the name for zed.Value slice.
```suggestion
func WriteZNG(t *testing.T, w io.Writer, vals []zed.Value) {
```